client/lib/purchases/index.js

```diff
@@ -27,6 +30,7 @@ function getPurchasesBySite( purchases ) {
 				domain: currentValue.domain,
 				id: currentValue.siteId,
 				name: currentValue.siteName,
+				slug: sites.getSite( currentValue.siteId ).slug,
```

Comment:
What do you think about removing this call to `SitesList` and updating this function to accept an array of sites as a parameter, like `getPurchasesBySite( purchases, sites )`? It would keep the function pure, make this easier to test, and would be clearer that information from `SitesList` is necessary in modules that use `getPurchasesBySite`.